fusionauth/resource_fusionauth_ldap_connector.go

```diff
@@ -0,0 +1,320 @@
+	}
+	if err := data.Set("requested_attributes", connector.RequestedAttributes); err != nil {
+		return diag.Errorf("connector.requested_attributes: %s", err.Error())
+	}
+	if err := data.Set("security_method", connector.SecurityMethod); err != nil {
+		return diag.Errorf("connector.security_method: %s", err.Error())
+	}
+	if err := data.Set("system_account_dn", connector.SystemAccountDN); err != nil {
+		return diag.Errorf("connector.system_account_dn: %s", err.Error())
+	}
+	if err := data.Set("system_account_password", connector.SystemAccountPassword); err != nil {
+		return diag.Errorf("connector.system_account_password: %s", err.Error())
+	}
+	return nil
+}
+
+func updateLDAPConnector(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	connector := buildLDAPConnector(data)
+
+	resp, faErrs, err := UpdateLDAPConnector(ctx, client.FAClient, data.Id(), LDAPConnectorRequest{Connector: connector})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func deleteLDAPConnector(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	id := data.Id()
+
+	resp, faErrs, err := client.FAClient.DeleteConnector(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+type LDAPConnectorRequest struct {
+	Connector fusionauth.LDAPConnectorConfiguration `json:"connector,omitempty"`
+}
+
+type LDAPConnectorResponse struct {
+	fusionauth.BaseHTTPResponse
+	Connector  fusionauth.LDAPConnectorConfiguration   `json:"connector,omitempty"`
+	Connectors []fusionauth.LDAPConnectorConfiguration `json:"connectors,omitempty"`
+}
+
+func (b *LDAPConnectorResponse) SetStatus(status int) {
+	b.StatusCode = status
+}
+
+func CreateLDAPConnector(ctx context.Context, client fusionauth.FusionAuthClient, connectorID string, request LDAPConnectorRequest) (*LDAPConnectorResponse, *fusionauth.Errors, error) {
+	return makeLDAPConnectorRequest(ctx, client, connectorID, request, http.MethodPost)
+}
+
+func RetrieveLDAPConnector(ctx context.Context, client fusionauth.FusionAuthClient, connectorID string) (*LDAPConnectorResponse, *fusionauth.Errors, error) {
+	return makeLDAPConnectorRequest(ctx, client, connectorID, LDAPConnectorRequest{}, http.MethodGet)
+}
+
+func UpdateLDAPConnector(ctx context.Context, client fusionauth.FusionAuthClient, connectorID string, request LDAPConnectorRequest) (*LDAPConnectorResponse, *fusionauth.Errors, error) {
+	return makeLDAPConnectorRequest(ctx, client, connectorID, request, http.MethodPut)
+}
+
+func makeLDAPConnectorRequest(ctx context.Context, client fusionauth.FusionAuthClient, connectorID string, request LDAPConnectorRequest, method string) (*LDAPConnectorResponse, *fusionauth.Errors, error) {
```

Comment:
In our dev environments, we have strict request parameters checks to make sure they explode at dev time, and not in prod. So you probably didn't see this, but this failed running acceptance tests against my dev env.

The problem is that GET requests sent from this function also pass in a request body of the empty `LDAPConnectorRequest` object, which isn't expected.

A dumb check on the method allowed it to pass (not to say that this the best fix):
```golang
        var body interface{}
	if method != http.MethodGet {
		body = request
	}

	restClient := client.Start(&resp, &errors)
	err := restClient.WithUri("/api/connector").
		WithUriSegment(connectorID).
		WithJSONBody(body).
		WithMethod(method).
		Do(ctx)
```